### PR TITLE
MM-10488 Adding /api/v4/users/usernames to loadtests.

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

File: terraform/cluster_deploy.go [hunk 1]
```diff
@@ -200,6 +200,7 @@ func deployToLoadtestInstance(instanceNum int, instanceAddr string, loadtestDist
 		".ConnectionConfiguration.SSHKey":               remoteSSHKeyPath,
 		".ConnectionConfiguration.MattermostInstallDir": "/opt/mattermost",
 		".ConnectionConfiguration.WaitForServerStart":   false,
+		".LoadtestEnviromentConfig.NumEmoji":            0,
```
Comment (crspeller) [inline terraform/cluster_deploy.go:203]: This is required because the emoji don't work with terraform.

File: loadtest/tests.go [hunk 2]
```diff
@@ -280,14 +280,20 @@ func actionGetChannel(c *EntityConfig) {
...
-			if rand.Float64() < c.LoadTestConfig.UserEntitiesConfiguration.CustomEmojiChance {
+			if rand.Float64() < c.LoadTestConfig.UserEntitiesConfiguration.CustomEmojiChance && c.LoadTestConfig.LoadtestEnviromentConfig.NumEmoji > 0 {
 				name := c.LoadTestConfig.LoadtestEnviromentConfig.PickEmoji()
 				if _, resp := c.Client.GetEmojiByName(name); resp.Error != nil {
 					mlog.Error("Unable to get emoji.", mlog.String("emoji_name", name), mlog.String("user_id", post.UserId), mlog.Err(resp.Error))
 				}
 			}
 		}
 	}
+	if rand.Float64() < c.LoadTestConfig.UserEntitiesConfiguration.NeedsProfilesChance {
```
Comment (lieut-data) [inline loadtest/tests.go]: In practice, is it just the one user that needs to be resolved? Or to ask this another way, should we sometimes loadtest querying for multiple users? I'm imagining a loop here that accumulates usernames, with the probability of starting an iteration still governed by the `NeedsProfilesChance`.

File: terraform/cluster_deploy.go (hunk 1 above)
Comment (lieut-data) [reply terraform/cluster_deploy.go:203]: Can you remind me what needs to be done to make that work? (for `mattermod`, if that ever gets back off the ground...)

**Review by lieut-data — APPROVED** [review]

File: loadtest/tests.go (hunk 2 above)
Comment (lieut-data) [reply loadtest/tests.go]: Also, based on my investigation into https://mattermost.atlassian.net/browse/MM-10483, it looks like I'll be adding similar code to cover `/api/v4/users/ids` and `/api/v4/users/status/ids`, so figuring out the pattern here will be tremendously helpful :)

File: terraform/cluster_deploy.go (hunk 1 above)
Comment (crspeller) [reply terraform/cluster_deploy.go:203]: We need to copy the emoji file so it can be picked up by the bulkload.

File: loadtest/tests.go (hunk 2 above)
Comment (crspeller) [reply loadtest/tests.go]: I calculated that it was the average number called.

Comment (lieut-data) [reply loadtest/tests.go]: Ah, ok, so the average number called by a single entity? My concern is that doesn't really reflect the client behaviour which might load a bunch of these in quick succession vs. distributing the queries out over the average.

**Review by jwilander — APPROVED** [review]

**Pull request merged**